test/plugins/smartQuotes.test.ts

```diff
@@ -0,0 +1,993 @@
+      title: 'last: odd quotes - 5',
+      start: ['aaa "bbb"| c "d" e " f'],
+      keysPressed: 'dil"',
+      end: ['aaa "|" c "d" e " f'],
+    });
+    newTest({
+      title: 'last: odd quotes - 6',
+      start: ['aaa "bbb" c |"d" e " f'],
+      keysPressed: 'dil"',
+      end: ['aaa "|" c "d" e " f'],
+    });
+    newTest({
+      title: 'last: odd quotes - 7',
+      start: ['aaa "bbb" c "|d" e " f'],
+      keysPressed: 'dil"',
+      end: ['aaa "|" c "d" e " f'],
+    });
+    newTest({
+      title: 'last: odd quotes - 8',
+      start: ['aaa "bbb" c "d|" e " f'],
+      keysPressed: 'dil"',
+      end: ['aaa "|" c "d" e " f'],
+    });
+    newTest({
+      title: 'last: odd quotes - 9',
+      start: ['aaa "bbb" c "d"| e " f'],
+      keysPressed: 'dil"',
+      end: ['aaa "bbb" c "|" e " f'],
+    });
+    newTest({
+      title: 'last: odd quotes - 10',
+      start: ['aaa "bbb" c "d" e |" f'],
+      keysPressed: 'dil"',
+      end: ['aaa "bbb" c "|" e " f'],
+    });
+    newTest({
+      title: 'last: odd quotes - 11',
+      start: ['aaa "bbb" c "d" e "| f'],
+      keysPressed: 'dil"',
+      end: ['aaa "bbb" c "|" e " f'],
+    });
+    newTest({
+      title: 'last: no space between - 1',
+      start: ['|"a""bbb"'],
+      keysPressed: 'dil"',
+      end: ['|"a""bbb"'],
+    });
+    newTest({
+      title: 'last: no space between - 2',
+      start: ['"|a""bbb"'],
+      keysPressed: 'dil"',
+      end: ['"|a""bbb"'],
+    });
+    newTest({
+      title: 'last: no space between - 3',
+      start: ['"a|""bbb"'],
+      keysPressed: 'dil"',
+      end: ['"a|""bbb"'],
+    });
+    newTest({
+      title: 'last: no space between - 4',
+      start: ['"a"|"bbb"'],
+      keysPressed: 'dil"',
+      end: ['"|""bbb"'],
+    });
+    newTest({
+      title: 'last: no space between - 5',
+      start: ['"a""|bbb"'],
+      keysPressed: 'dil"',
+      end: ['"|""bbb"'],
+    });
+    newTest({
+      title: 'last: no space between - 6',
+      start: ['"a""bbb|"'],
+      keysPressed: 'dil"',
+      end: ['"|""bbb"'],
+    });
+    newTest({
+      title: 'last: no space between - 7',
+      start: ['"a""bbb"| '],
+      keysPressed: 'dil"',
+      end: ['"a""|" '],
+    });
+  });
+
+  suite('smartQuotes.breakThroughLines = true', () => {
+    setup(async () => {
+      const configuration = new Configuration();
+      configuration.smartQuotes.enable = true;
+      configuration.smartQuotes.breakThroughLines = true;
+      await setupWorkspace(configuration, '.js');
+    });
+
+    teardown(cleanUpWorkspace);
+
+    // test next
+    newTest({
+      title: 'next: should go next line - 1',
+      start: [
+        'aaa "bbb" c', //
+        'd |"e" f', //
+        'g "h" k',
+      ],
+      keysPressed: 'din"',
+      end: [
+        'aaa "bbb" c', //
+        'd "e" f', //
+        'g "|" k',
+      ],
+    });
+    newTest({
+      title: 'next: should go next line - 2',
+      start: [
+        'aaa "bbb" c', //
```

Comment:
What's with all the empty comments?

Reply:
Prettify automatically joins the lines, and I think it's make it hard to read the code otherwise.

We can do this instead, if you prefer
```suggestion
        'aaa "bbb" c', // linebreak
```